lib/Transport.js

```diff
@@ -636,6 +653,11 @@ function lowerCaseHeaders (oldHeaders) {
   return newHeaders
 }
 
+function validateMemoryPercentage (percentage) {
+  if (percentage < 0 || percentage > 1) return 1.0
```

Comment:
Seems like percentage could be nullable right but it has a default value so will it ever be null?

Reply:
It wont be null since we have the default value assgined.

Reply:
```
  memoryCircuitBreaker?: {
    enabled: boolean | false;
    maxPercentage?: number | 1.0;
  }
```

I think I led you astray. Your object's properties consist of union types. So `enabled `being `boolean | false` is redundant because we say enabled can only be of type `boolean`. Where as `maxPercentage` which has is optional with the `?` has the redundant `number | 1.0` by it an optional type that is actually `number | undefined` so someone can create a memoryCircuitBreaker   and it won't have default values unless there's a file that you are defining defaults here. But in the end this is editing a JS file so these values can be undefined even if you define a type.

Reply:
I think I confused the use of `|` in javascript type definition. I thought it will do similar thing as logical operator `value || defaultValue`. I intentionally tested this scenario and passed because In OSD, I have  default value assigned for `memoryCircuitBreaker` object. Thanks for pointing out. will update